src/auto_scheduler/measure_record.cc

```diff
@@ -182,6 +184,10 @@ struct Handler<::tvm::auto_scheduler::SearchTaskNode> {
     reader->Read(&str_value);
     data->target = ::tvm::Target(str_value);
     s = reader->NextArrayItem();
+    ICHECK(s);
+    reader->Read(&int_value);
+    data->layout_rewrite_option = ::tvm::auto_scheduler::LayoutRewriteOption(int_value);
+    s = reader->NextArrayItem();
```

Comment:
Why cannot we make it optional following the pattern below?

Ideally, we should be able to do parsing according to the version number. It is easy to do this with python's json library, but it seems it is not easy to do this with the our json parser from dmlc-core. Because we write the version number as the last field, we cannot get it when parsing the fileds before it.

If we cannot keep backward compatability, we should issue a waring message and provide an upgrade script.

Reply:
The problem seems not only about the compatability.
The code above:
https://github.com/apache/tvm/blob/0b2ab56705a1c26aef402d1c0d48a887fc76dbd3/src/auto_scheduler/measure_record.cc#L166-L168
allows different log format even when the log version is v0.4.

Reply:
I see. We should fix this by
```
 if (data.target_host.defined()) { 
   writer->WriteArrayItem(data.target_host->str()); 
 } else {
   writer->WriteArrayItem("");
 } 
```
We can make sure that new writer always writes all fields into the file. But the new reader can read old logs with some fileds missing.

Reply:
Done.